cli/azd/cmd/init.go

```diff
@@ -125,11 +125,8 @@ func (i *initAction) Run(ctx context.Context) (*actions.ActionResult, error) {
 		return nil, fmt.Errorf("getting cwd: %w", err)
 	}
 
-	azdCtx, err := i.lazyAzdCtx.GetValue()
-	if err != nil {
-		azdCtx = azdcontext.NewAzdContextWithDirectory(wd)
-		i.lazyAzdCtx.SetValue(azdCtx)
-	}
+	azdCtx := azdcontext.NewAzdContextWithDirectory(wd)
+	i.lazyAzdCtx.SetValue(azdCtx)
```

Comment:
This does not feel like the correct fix.  If a command asked for an azdCtx to be injected we should be able to trust that it was constructed based on our supported scenarios.

IMO - the fix should be made during the construction of the instance in the DI container to take check for the existence of the `--cwd` flag.

Reply:
This change has nothing do with `--cwd` or flag resolution. That part is working as expected. If you believe otherwise, let's have that discussion here. 

Focusing on the real behavior here, there's two ways of constructing an `azdContext`:

1. `NewAzdContext` - resolves based on the nearest parent directory
2. `NewAzdContextWithDirectory` - creates `azdContext` based on the specified directory

What should be the right behavior here? My belief is that running `azd init` in any directory, should simply initialize in that directory, not any parent directory.

Reply:
Now the second part is how should we architect for such a behavior, especially in a DI world? Certainly, the mutation looks weird here, including when introduced initially in #2765. Personally, I don't think it's right that `azdContext` should be injected at all in things like `envManager`.

A backstory here is that I did initially write `azdContext` to be resolvable, which the idea that the action is the "application", and the command can easily be constructed with the context loaded, but it is a scoped variable that is passed around in the main logic (which is `action.Run`).

If we believe in the hypothesis that we should move `azdContext` away from being curried in services/managers, I am happy to make the change there. But given that there wasn't any historical context around this specific change, I wanted to invite that discussion, which I'm happy that we're discussing about it now.

Reply:
Another way of thinking about in a DI world is that `init` updates the registration to resolve `azdContext` differently. This would either require introducing an interface, or reregistering `azdContext` which can be susceptible to registration ordering. Personally, I feel that this is trading off [locality of behavior](https://htmx.org/essays/locality-of-behaviour/) for more separations of concerns, but it doesn't provide the pragmatism of working in a fast-paced environment of azd.

Reply:
Can we have one `azdContext` and one `azdInitialContext` ?  

It seems like one should be wrapped by the other, but this could make it clear that the azdContext is created differently when running `azd init` than when running other commmands

Reply:
But `azd init` implementation should not take the responsibility of creating/overriding what DI should do.

Reply:
@vhvb1989 That would end up creating an interface based on my knowledge, which is the option I proposed above.

Also, what is an `azdInitialContext` outside of `azd init`?

Reply:
Yeah. 
On second thought, `azd init` should probably not require an azdContext...  as there's not any context yet.
Maybe we need something else, like a `EnvTemplate` or `envBootstrap`, which is injected by DI, and contains the path, and any other ingredients which might affect creating a new azd env.

Then `azd init` would just do the job, but it would never try to figure the inputs on its own.

I recently added a bootstrap-strategy for the env-config..  That would be part of the env-template/component created by DI and injected to azd init action

Reply:
@vhvb1989 To be precise, it's managers like `envManager` that needs the `azdContext`, not `init`. Which comes back to the following comment posted previously:

If we believe in the hypothesis that we should move `azdContext` away from being curried in services/managers, I am happy to make the change there. But given that there wasn't any historical context around this specific change, I wanted to invite that discussion, which I'm happy that we're discussing about it now.

Reply:
Will move discussion to #3068 to keep this going while fixing the behavioral issue.